thirdeye-scheduler/src/main/java/ai/startree/thirdeye/scheduler/TaskCronSchedulerRunnable.java

```diff
@@ -97,43 +119,99 @@ public void shutdown() throws SchedulerException {
   }
 
   private void updateSchedules() throws SchedulerException {
+
+    final Supplier<HashMap<String, Boolean>> namespaceToQuotaExceededMap = scheduledRefreshSupplier(
+        this::getNamespaceToQuotaExceededMap, Duration.ofSeconds(namespaceQuotaMapRefreshDelay));
+
     // TODO CYRIL scale - loading all entities is expensive - only the id and the cron are necessary - requires custom SQL (JOOQ)
     // FIXME CYRIL SCALE - No need for a strong isolation level here - the default isolation level lock all entities until the query is finished, blocking progress of tasks and potentially some update/delete operations in the UI.
     //   dirty reads are be fine, this logic runs every minute
     //   also only fetch only active entities directly and remove is active from Schedulable interface 
     final List<E> allEntities = entityDao.findAll();
 
     // schedule active entities
-    allEntities.forEach(this::schedule);
+    allEntities.forEach(e -> schedule(e, namespaceToQuotaExceededMap.get()));
 
     // cleanup schedules of deleted and deactivated entities
+    // or entities whose workspace has exceeded quotas
     final Map<Long, E> idToEntity = allEntities.stream()
         .collect(Collectors.toMap(AbstractDTO::getId, e -> e));
     final Set<JobKey> scheduledJobKeys = scheduler.getJobKeys(groupMatcher);
     for (final JobKey jobKey : scheduledJobKeys) {
       try {
         final Long id = getIdFromJobKey(jobKey);
         final E entity = idToEntity.get(id);
+        final String entityNamespace = entity.namespace();
         if (entity == null) {
           log.info("{} with id {} does not exist anymore. Stopping the scheduled {} job.",
               entityName, id, taskType);
           stopJob(jobKey);
         } else if (!isActive(entity)) {
           log.info("{} with id {} is deactivated. Stopping the scheduled {} job.", entityName, id, taskType);
           stopJob(jobKey);
+        } else if (namespaceToQuotaExceededMap.get().getOrDefault(entityNamespace, false)) {
```

Comment:
let's re-use the same cached map in a call to `updateSchedules`
here it's called line 133 and line 152
having the cached value changed during 1 call would be a bit confusing when troubleshooting.

so only call 
```
namespaceToQuotaExceededMap.get()
```
once, line 132